### PR TITLE
Refactor math and add support for Greek characters

### PR DESCRIPTION
**Pull request by wismill**
This PR:
* split the `Math` module into submodules
* add support for Greeks symbols
* changes the behavior of the functions converting chars: 
    * in `Data.Char.Math` all supported symbols are converted
    * the `Latin`, `Greek` and `digit` variants in their respective modules only convert the corresponding symbols.

### CODE REVIEW

**Comment by wismill** [comment]
Not all the documentation is updated, thus the "WIP" status. I am not sure I will have the time to update it.

**Comment by KommuSoft** [comment]
Ok, I will try to find some time to finish the documentation. Thanks :+1:

**Comment by wismill** [comment]
I think it would be best to add a definition of the ranges latin, greek and digits at the top of the modules rather than to duplicate it in every function.

**Comment by wismill** [comment]
Ready for review

**Comment by wismill** [comment]
Hi, let me know if something need to be modified.

**Comment by KommuSoft** [comment]
Sorry did not have much time lately, will try to review it today.

**Comment by KommuSoft** [comment]
Is it possible to also update the `README.md`? (if it is too much work, I will update it).

File: src/Data/Char/Core.hs [hunk 1]
```diff
@@ -211,6 +211,12 @@ isAsciiAlpha x = isAscii x && isAlpha x
 isAsciiAlphaNum :: Char -> Bool
 isAsciiAlphaNum x = isAscii x && isAlphaNum x
 
+-- | Checks if a charcter is a basic /greek alphabetic/ character or a greek-like symbol. The characters
+-- @ΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ∇ϴαβγδεζηθικλμνξοπρςστυφχψω∂ϵϑϰϕϱϖ@ satisfy this
+-- predicate.
+isGreek :: Char -> Bool
+isGreek = (`elem` ("ΑΒΓΔΕΖΗΘΙΚΛΜΝΞΟΠΡΣΤΥΦΧΨΩ∇ϴαβγδεζηθικλμνξοπρςστυφχψω∂ϵϑϰϕϱϖ" :: String))
```
[the remark this answers is not in the document]
Comment (wismill) [reply src/Data/Char/Core.hs]: Ok, done

**Comment by wismill** [comment]
> Is it possible to also update the `README.md`? (if it is too much work, I will update it).

I think it is good to keep it light. I would not expand the `Data.Char.Math`.

**Pull request merged**